### DISCUSSION TITLE
Negating query

### DISCUSSION THREAD

**Discussion by lumio** [typst/typst#3904] [Q&A] [answered]
Hello everyone!

Maybe I have overlooked something, but I can't figure out how to negate a query when filtering figures to be outlined.

I am trying to achieve something like this:

```typst
= Figures
#outline(title: none, target: figure.whereNot(supplement: [Attachment]))
```

How can I filter out certain fields?

Thank you for your help!

**Comment by laurmaedje** [comment]
It's unfortunately not possible currently. :/

You'd have to write an or selector for all figure kinds you want to include explicitly.

**Reply by lumio** [reply]
Thanks for your reply. The only workaround I can think of right now is to manually chain all existing supplements. However, this isn't foolproof either as it is language specific.

So what I did for now was to [search for all supplement translations](https://github.com/search?q=repo%3Atypst%2Ftypst+%22Lang%3A%3AGERMAN+%3D%3E%22&type=code) and then do this here:

```typst
  #outline(
    title: none,
    target: figure.where(supplement: [Listing])
      .or(figure.where(supplement: [Abbildung]))
      .or(figure.where(supplement: [Gleichung]))
      .or(figure.where(supplement: [Tabelle]))
  )
```

If there isn't a feature request yet, I'll create one :)

Thanks again @laurmaedje

**Reply by laurmaedje** [reply, marked as answer]
It's better to select on the `kind` rather than the `supplement`. Then, it will not be language-dependant:

```
  #outline(
    title: none,
    target: figure.where(kind: raw)
      .or(figure.where(kind: image)
      .or(figure.where(kind math.equation)
      .or(figure.where(kind: table)
  )
```
